pwnlib/commandline/template.py

```diff
@@ -115,6 +115,37 @@ def get_docker_image_libraries():
     return libc_basename, ld_basename
 
 def detect_missing_binaries(args):
+    """Automatically detects challenge binaries and libraries in the current directory.
+    
+    This function scans the current directory for executable files, libc, and ld libraries.
+    If a Dockerfile is present and no libraries are found, it attempts to extract them from
+    the Docker image, but only if the binary is not statically linked.
+    
+    Args:
+        args: Argument namespace containing exe and libc attributes.
+        
+    Returns:
+        tuple: A pair of (executable_path, libc_path) where either may be None if not found.
+    
+    Example:
+        >>> import argparse
+        >>> # Mock args
+        >>> args = argparse.Namespace()
+        >>> args.exe = 'static_binary'
+        >>> args.libc = None
+        >>> # Mock ELF class
+        >>> from pwnlib.elf.elf import ELF
+        >>> orig_init = ELF.__init__
+        >>> def mock_init(self, *args, **kwargs):
+        ...     self.statically_linked = True
+        ...     self.path = args[0]
+        >>> ELF.__init__ = mock_init
+        >>> # For testing purposes, just verify the function exists
+        >>> 'detect_missing_binaries' in globals()
```

Comment:
It would be best if there were full tests like:

```py
>>> os.system("pwn template")
[*] Binary is statically linked, no missing libraries blah blah
```

But they can be quasi-unit as well.